numba/npyufunc/_internal.c

```diff
@@ -104,6 +105,557 @@ PyTypeObject PyUFuncCleaner_Type = {
     0,                                          /* tp_version_tag */
 };
 
+/* ______________________________________________________________________
+ * DUFunc: A call-time (hence dynamic) specializable ufunc.
+ */
+
+typedef struct {
+    PyObject_HEAD
+    PyObject      * dispatcher;
```

Comment:
After thinking about this for a bit, I'm going to hold to API-cleanliness here.  The dispatcher object holds information used in dufunc_inti() that could be teased out by the concrete subclass, but then we end up with very C-like constructor calls.